packages/xrpl/src/models/utils/flags.ts

```diff
@@ -63,37 +61,61 @@ const txToFlag = {
   XChainModifyBridge: XChainModifyBridgeFlags,
 }
 
+function istxToFlagKey(
+  transactionType: string,
+): transactionType is keyof typeof txToFlag {
+  return transactionType in txToFlag
+}
+
 /**
  * Sets a transaction's flags to its numeric representation.
  *
+ * @deprecated
+ * This utility function is deprecated.
+ * Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.
+ *
  * @param tx - A transaction to set its flags to its numeric representation.
  */
 export function setTransactionFlagsToNumber(tx: Transaction): void {
-  if (tx.Flags == null) {
-    tx.Flags = 0
-    return
+  // eslint-disable-next-line no-console -- intended deprecation warning
+  console.warn(
+    'This function is deprecated. Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.',
+  )
+
+  if (tx.Flags) {
+    // eslint-disable-next-line no-param-reassign -- intended param reassign in setter, retain old functionality for compatibility
+    tx.Flags = convertTxFlagsToNumber(tx)
+  }
+}
+
+/**
+ * Returns a transaction's flags as its numeric representation.
+ *
+ * @param tx - A transaction to parse flags for
+ * @returns A numerical representation of a transaction's flags
+ */
+export function convertTxFlagsToNumber(tx: Transaction): number {
+  if (!tx.Flags) {
+    return 0
   }
   if (typeof tx.Flags === 'number') {
-    return
+    return tx.Flags
   }
 
-  tx.Flags = txToFlag[tx.TransactionType]
-    ? convertFlagsToNumber(tx.Flags, txToFlag[tx.TransactionType])
-    : 0
-}
+  if (istxToFlagKey(tx.TransactionType)) {
+    const flagEnum = txToFlag[tx.TransactionType]
+    return Object.keys(tx.Flags).reduce((resultFlags, flag) => {
+      if (flagEnum[flag] == null) {
+        throw new ValidationError(
+          `flag ${flag} doesn't exist in flagEnum: ${JSON.stringify(flagEnum)}`,
```

Comment:
Better to not reference `flagEnum` variable within the function. Users who get this exception won't understand the variable unless they inspect the function.

```suggestion
          Invalid flag ${flag}. Valid flags are ${JSON.stringify(flagEnum)}`,
```